rfcs/45-stageless.md

```diff
@@ -496,232 +465,269 @@ They are evaluated a single time during each schedule pass, and the result appli
-   2. This can be easily added later, or third-party plugins can create their own abstraction.
-4. When a state transition occurs, the systems in the while-active set are no longer applied immediately.
-   1. Instead, the normal flow of logic continues.
-   2. Users can duplicate critical logic in the while-active and on-enter collections.
-   3. Similarly, arbitrarily long chains of state transitions are no longer be processed in the same schedule pass.
-   4. However, users can add as many copies of the `flush_state` system as they would like, and loop it within an exclusive system.
+    - This also removes "in-stack" and related system groups / logic.
+    - With the simplified run criteria, the stack is trivial to re-implement, but preferably a third-party plugin does it.
+    - The `apply_state_transition` system performs a single transition, doesn't do an endless cycle in one schedule pass.
+4. States no longer have an "on update".
+    - If a system must run on the same frame as the transition, it has to be in "on enter" and "on exit" (or your own DIY equivalent).
 
 ## Rationale and alternatives
 
-### Doesn't set-configuration allow you to configure systems added by third-party plugins?
+### Why did you call them "sets" instead of "labels" or "categories"?
 
-Yes. This is a large part of the point.
+The word "set" better describes what labeling actually means for scheduling. Sets aren't just an abstract tag, they have a "physical" location in the overall graph (a set is a "supernode" that encapsulates all the nodes inside it). Likewise, sets that intersect (share nodes) can't be made to depend on each other.
 
-The difficulties configuring plugins is a source of [significant user pain](https://github.com/bevyengine/bevy/issues/2160).
-Plugin authors cannot know all of the needs of the end user's app, and so some degree of control must be handed back to the end users.
+### If users can configure system sets, can they mess with systems added by third-party plugins?
 
-This control is carefully limited though: systems can only be configured if they are part of a public set, and all of the systems under a common set are configured as a group.
-In addition, configuration defined by a private set (or directly on systems) cannot be altered, extended or removed by the app: it cannot be named, and so cannot be altered.
+Yes, somewhat, and it's intentional.
 
-This limitation allows plugin authors to carefully design a public API, ensure critical invariants hold, and make serious internal changes without breaking their dependencies (as long as the public sets and their meanings are relatively stable).
+Not being able to schedule plugin systems is a source of [significant user pain](https://github.com/bevyengine/bevy/issues/2160). Instead of the current status quo, we thought a better pattern would be a plugin exporting its top-level set label (at the very least) and the user being responsible for deciding where to put it.

```

Comment:
```suggestion
Not being able to schedule plugin systems is a source of [significant user pain](https://github.com/bevyengine/bevy/issues/2160).
Instead of the current status quo, we thought a better pattern would be a plugin exporting its top-level set label (at the very least) and the user being responsible for deciding where to put it.
```